rsts/deployment/plugin_setup/aws/batch.rst

```diff
@@ -0,0 +1,155 @@
+
+AWS Batch Setup for Map Tasks
+-----------------------------
+
+AWS Batch enables developers, scientists, and engineers to easily and efficiently run hundreds of thousands of batch
+computing jobs on AWS.
+
+Flyte abstracts away the complexity of integrating AWS Batch into users' workflows. It takes care of packaging inputs,
+reading outputs, scheduling map tasks, leveraging AWS Batch Job Queues to distribute the load and coordinate priorities.
+
+1. Set-up AWS Batch
+
+   At the end of this step, the AWS Account should have a configured Compute Environment and one or more AWS Batch Job Queues.
+   Follow the guide `Running batch jobs at scale for less <https://aws.amazon.com/getting-started/hands-on/run-batch-jobs-at-scale-with-ec2-spot/>`_.
+
+2. Modify Users' AWS IAM Role trust policy document
+
+   When running Workflows in Flyte, users have the option to specify a K8s Account and/or an IAM Role to run as. For AWS Batch, an IAM Role must be specified.
+   For every one of these IAM Roles, modify the trust policy to allow ECS to assume the role.
+   Follow the guide `AWS Batch Execution IAM role <https://docs.aws.amazon.com/batch/latest/userguide/execution-IAM-role.html>`_.
+
+3. Modify System's AWS IAM Role policies
+
+   The recommended way of assigning permissions to flyte components is using `OIDC <https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html>`_.
+   This involves assigning an IAM Role for every service account used. You will need to find the IAM Role assigned to the flytepropeller's kubernetes service account.
+   Then modify the policy document to allow the role to pass other roles to AWS Batch.
+   Follow the guide: `Granting a user permissions to pass a role to an AWS service <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_passrole.html>`_.
+
+4. Update Flyte Admin's Config
+
+   Flyte Admin needs to be made aware of all the AWS Batch Job Queues and how the system should distribute the load onto them.
+   The simplest setup looks something like this:
+
+   .. code-block:: yaml
+
+      flyteadmin:
+        roleNameKey: "eks.amazonaws.com/role-arn"
+      queues:
+        # A list of items, one per AWS Batch Job Queue.
+        executionQueues:
+          # The name of the job queue from AWS Batch
+          - dynamic: "tutorial"
+            # A list of tags/attributes that can be used to match workflows to this queue.
+            attributes:
+              - default
+        # A list of configs to match project and/or domain and/or workflows to job queues using tags.
+        workflowConfigs:
+          # An empty rule to match any workflow to the queue tagged as "default"
+          - tags:
+              - default
+
+   If you are using Helm, this block can be added under `configMaps.adminServer` section `here <https://github.com/flyteorg/flyte/blob/master/charts/flyte/values.yaml#L526-L527>`_.
+
+   An example of a more complex matching config:
```

Comment:
not sure if this is overkill here, but should we link to how you can set specific queue tags based on project/domain etc

- admin: background: https://docs.flyte.org/en/latest/concepts/admin.html#matchable-resources
- proto def: https://docs.flyte.org/projects/flyteidl/en/latest/protos/docs/admin/admin.html#executionqueueattributes
- flytectl: https://docs.flyte.org/projects/flytectl/en/stable/gen/flytectl_update_execution-queue-attribute.html#id1

Reply:
Done! didn't put in the proto def... please let me know if you feel like I should.